cajita/unit_test/tstBovWriter.hpp

```diff
@@ -64,53 +64,59 @@ void writeTest3d()
 
     // Field data values.
     double pi2 = 8.0 * atan( 1.0 );
-
-    // Create a scalar cell field and fill it with data.
-    auto cell_layout = createArrayLayout( global_grid, 0, 1, Cell() );
-    auto cell_field =
-        createArray<double, TEST_DEVICE>( "cell_field_3d", cell_layout );
-    auto cell_data = cell_field->view();
-    Kokkos::parallel_for(
-        "fill_cell_field",
-        createExecutionPolicy(
-            cell_layout->localGrid()->indexSpace( Own(), Cell(), Local() ),
-            TEST_EXECSPACE() ),
-        KOKKOS_LAMBDA( const int i, const int j, const int k ) {
-            double xarg = double( off_i + i ) / num_cell_dev[0];
-            double yarg = double( off_j + j ) / num_cell_dev[1];
-            double zarg = double( off_k + k ) / num_cell_dev[2];
-            cell_data( i, j, k, 0 ) =
-                1.0 + fabs( cos( pi2 * xarg ) * cos( pi2 * yarg ) *
-                            cos( pi2 * zarg ) );
-        } );
-
-    // Create a vector node field and fill it with data.
-    auto node_layout = createArrayLayout( global_grid, 0, 3, Node() );
-    auto node_field =
-        createArray<double, TEST_DEVICE>( "node_field_3d", node_layout );
-    auto node_data = node_field->view();
-    Kokkos::parallel_for(
-        "fill_node_field",
-        createExecutionPolicy(
-            node_layout->localGrid()->indexSpace( Own(), Node(), Local() ),
-            TEST_EXECSPACE() ),
-        KOKKOS_LAMBDA( const int i, const int j, const int k ) {
-            double xarg = double( off_i + i ) / num_cell_dev[0];
-            double yarg = double( off_j + j ) / num_cell_dev[1];
-            double zarg = double( off_k + k ) / num_cell_dev[2];
-            node_data( i, j, k, Dim::I ) = 1.0 + fabs( cos( pi2 * xarg ) );
-            node_data( i, j, k, Dim::J ) = 1.0 + fabs( cos( pi2 * yarg ) );
-            node_data( i, j, k, Dim::K ) = 1.0 + fabs( cos( pi2 * zarg ) );
-        } );
-
-    // Gather the node data.
-    auto node_halo = createHalo( NodeHaloPattern<3>(), 0, *node_field );
-    node_halo->gather( TEST_EXECSPACE(), *node_field );
-
-    // Write the fields to a file.
-    Experimental::BovWriter::writeTimeStep( 302, 3.43, *cell_field );
-    Experimental::BovWriter::writeTimeStep( 1972, 12.457, *node_field );
-
+    {
+        // Create a scalar cell field and fill it with data.
+        auto cell_layout = createArrayLayout( global_grid, 0, 1, Cell() );
+        auto cell_field =
+            createArray<double, TEST_DEVICE>( "cell_field_3d", cell_layout );
+        auto cell_data = cell_field->view();
+
+#if ( defined __SYCL_DEVICE_ONLY__ )
+        using Kokkos::Experimental::cos;
+        using Kokkos::Experimental::fabs;
+#endif
```

Comment:
The whole point of providing math functions in Kokkos was that you don't do things like that in user code.
Just use them unconditionally.

Reply:
I meant to reply here - as we discussed, until we require the next Kokkos release (with math function support) I'll leave this